include/aws/http/request_response.h

```diff
@@ -296,6 +296,60 @@ struct aws_http_request_handler_options {
     aws_http_on_stream_complete_fn *on_complete;
 };
 
+struct aws_http1_stream_chunk;
+
+/**
+ * Signature definition of the function which will be invoked, if provided by the caller, after a chunked transfer
+ *  encoding chunk has been written to an HTTP/1.1 connection.
+ */
+typedef void aws_http1_stream_write_chunk_complete_fn(struct aws_http1_stream_chunk *chunk, void *user_data);
```

Comment:
I'm on the fence about passing `struct aws_http1_stream_chunk` here.
The only thing a completion callback ever *really* needs to pass is user_data, where the user can store whatever they need to know when the call completes.

The `chunk` has everything in it the user might want to know, but also internal detail stuff like `node`. I'd kind of rather just hide that struct in a private header as an implementation detail.

Reply:
I agree. Will remove. If they caller needs anything from that struct, they can save the pointer to it themselves.